src/librustc_borrowck/diagnostics.rs

```diff
@@ -429,6 +429,96 @@ You can find more information about borrowing in the rust-book:
 http://doc.rust-lang.org/stable/book/references-and-borrowing.html
 "##,
 
+E0509: r##"
+This error occurs when an attempt is made to move out of a container that
+defines the `Drop` trait. For example:
+
+```compile_fail
+struct FancyNum {
+    num: usize
+}
+
+struct DropStruct {
+    fancy: FancyNum
+}
+
+impl Drop for DropStruct {
+    fn drop(&mut self) {
+        // Destruct DropStruct, possibly using FancyNum
+    }
+}
+
+fn main() {
+    let drop_struct = DropStruct{fancy: FancyNum{num: 5}};
+    let fancy_field = drop_struct.fancy; // Error E0509
+    println!("Fancy: {}", fancy_field.num);
+    // implicit call to `drop_struct.drop()` as drop_struct goes out of scope
+}
+```
+
+Here, we tried to move a field out of a struct of type `DropStruct` which
+implements the `Drop` trait.
+
+`Drop` types have an implicit destructor that gets called when they go out of
```

Comment:
This expression is a bit unwieldy. The correct explanation would be that a struct cannot be dropped if one of its field(s) is borrowed.

Also, I don't like the expression "`Drop` types". I prefer to use "Types which implement `Drop` trait".